openedx/core/djangoapps/models/course_details.py

```diff
@@ -271,6 +275,7 @@ def update_from_json(cls, course_key, jsondict, user):  # pylint: disable=too-ma
                 and jsondict['self_paced'] != descriptor.self_paced):
             descriptor.self_paced = jsondict['self_paced']
             dirty = True
+            is_pacing_changed = True
```

Comment:
we should fire this signal when changes are actually reflected in db – i.e. after `module_store.update_item(descriptor, user.id)`